plugin/src/index.ts

```diff
@@ -1,93 +1,45 @@
-import {
-  withAndroidManifest,
-  createRunOncePlugin,
-  ExportedConfigWithProps,
-  ExportedConfig,
-} from '@expo/config-plugins';
-
-// eslint-disable-next-line import/no-commonjs, @typescript-eslint/no-var-requires
-const pkg = require('../../package.json');
-
-/**
- * @type {import('./types').ManifestQueries}
- * what we are trying to add:
- * <queries>
-    <package android:name="com.facebook.katana"/>
-    <package android:name="com.instagram.android"/>
-    <package android:name="com.twitter.android"/>
-    <package android:name="com.zhiliaoapp.musically"/>
-    <intent></intent>
-      <action android:name="android.intent.action.VIEW"/>
-      <category android:name="android.intent.category.BROWSABLE"/>
-      <data android:scheme="https"/>
-    </intent>
-  </queries>
- */
-
-/**
- * @param {import('@expo/config-plugins').ExportedConfig} config
- */
-const withAndroidManifestService = (config: ExportedConfig, props: WithSocialShareProps) => {
-  return withAndroidManifest(config, (config: ExportedConfigWithProps) => {
-    config.modResults.manifest = {
-      ...config.modResults.manifest,
-      queries: {
-        package: props?.android?.map((social) => ({
-          $: {
-            'android:name': social,
-          },
-        })),
-        intent: [
-          {
-            action: {
-              $: {
-                'android:name': 'android.intent.action.VIEW',
-              },
-            },
-            category: {
-              $: {
-                'android:name': 'android.intent.category.BROWSABLE',
-              },
-            },
-            data: {
-              $: {
-                'android:scheme': 'https',
-              },
-            },
-          },
-        ],
+import { ExportedConfig } from '@expo/config-plugins';
+import { withBuildProperties } from 'expo-build-properties';
+
+export default (
+  config: ExportedConfig,
+  props: {
+    enableBase64ShareAndroid?: boolean;
+    android?: string[];
+    ios?: string[];
+  },
+) => {
+  return withBuildProperties(
+    {
+      ...config,
+      android: {
+        ...config.android,
+        ...(props.enableBase64ShareAndroid
+          ? {
+              permissions: [
+                ...(config.android?.permissions ?? []),
+                'android.permission.WRITE_EXTERNAL_STORAGE',
+              ],
```

Comment:
Should we remove duplicates in case `WRITE_EXTERNAL_STORAGE` is already passed as a permission?
```suggestion
              permissions: [...new Set([
                ...(config.android?.permissions ?? []),
                'android.permission.WRITE_EXTERNAL_STORAGE',
              ])],
```

Reply:
We could. But I think expo should already be doing that when computing the end result anyway